File: MATTHEW/leetcode/AIcoding/4.GPT-4o/rust_solutions/LCR 016-wtcaE1.rs
impl Solution {
    pub fn length_of_longest_substring(s: String) -> i32 {
        let mut char_index = std::collections::HashMap::new();
        let mut max_length = 0;
        let (mut start, n) = (0, s.len());

        for (i, c) in s.chars().enumerate() {
            if let Some(&last_index) = char_index.get(&c) {
                start = start.max(last_index + 1);
            }
            char_index.insert(c, i);
            max_length = max_length.max(i - start + 1);
        }

        max_length as i32
    }
}

File: leetcode/AIcoding/2.Gemini-2.0/c_solutions/LCR 016-wtcaE1.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int lengthOfLongestSubstring(char * s){
    int n = strlen(s);
    int maxLength = 0;
    int start = 0;
    int end = 0;
    int charIndex[256];
    for (int i = 0; i < 256; i++) {
        charIndex[i] = -1;
    }

    while (end < n) {
        if (charIndex[s[end]] != -1 && charIndex[s[end]] >= start) {
            start = charIndex[s[end]] + 1;
        }
        charIndex[s[end]] = end;
        maxLength = (end - start + 1) > maxLength ? (end - start + 1) : maxLength;
        end++;
    }

    return maxLength;
}